deerlab/solvers.py

```diff
@@ -917,9 +918,7 @@ def ymodel(n):
     stats = _goodness_of_fit_stats(ys,yfits,noiselvl,Ndof)
 
     # Display function
-    def plotfcn(show=False,axis=None,xlabel=None):
-        fig = _plot(ys,yfits,yuq,axis,xlabel)
-        return fig
+    plotfcn = partial(_plot,ys,yfits,yuq)

```

Comment:
Is `partial` really needed here? The short `def` seems fine. What does `partial` enable that `def` doesn't?

Reply:
Yes it is. Otherwise one would have to define `plotfcn` as a local function with `ys` and `yfits` in its local environment (as we had before), which is not pickable.